test/performance/alignment/edit_distance_benchmark_template.hpp

```diff
@@ -0,0 +1,98 @@
+//  seqan3 edit distance pairwise alignment benchmarks
+// ----------------------------------------------------------------------------
+
+//!\brief This directly benchmarks the alignment algorithm without going through the whole align_pairwise function
+template <typename sequence_pair_generator_t, typename align_configs_t>
+void seqan3_align_pairwise_edit_distance_benchmark(benchmark::State & state,
+                                                   sequence_pair_generator_t sequence_pair_generator,
+                                                   align_configs_t && edit_distance_cfg)
+{
+    using seqan3::test::edit_distance_algorithm;
+
+    auto sequence_pair_or_pairs = sequence_pair_generator(state);
+    constexpr bool is_collection = sequence_pair_generator.is_collection;
+    using sequence_t = typename sequence_pair_generator_t::sequence_t;
+
+    int64_t total = 0;
+
+    auto algorithm = edit_distance_algorithm::select<sequence_t, sequence_t>(edit_distance_cfg);
+    for (auto _ : state)
+    {
+        if constexpr (is_collection)
+        {
+            for (auto && [sequence1, sequence2] : sequence_pair_or_pairs)
+                total += algorithm(sequence1, sequence2, edit_distance_cfg).score();
+        }
+        else
+        {
+            auto & [sequence1, sequence2] = sequence_pair_or_pairs;
+            total += algorithm(sequence1, sequence2, edit_distance_cfg).score();
+        }
+    }
+
+    auto view_adaptor = std::conditional_t<is_collection, decltype(std::views::all), decltype(std::views::single)> {};
+    state.counters["cells"] = seqan3::test::pairwise_cell_updates(view_adaptor(sequence_pair_or_pairs),
+                                                                  edit_distance_cfg);
+    state.counters["CUPS"] = seqan3::test::cell_updates_per_second(state.counters["cells"]);
+    state.counters["total"] = total;
+}
+
+#ifdef SEQAN3_HAS_SEQAN2
+
+// ----------------------------------------------------------------------------
+//  seqan2 edit distance pairwise alignment benchmarks
+// ----------------------------------------------------------------------------
+
+template <typename sequence_pair_generator_t, typename align_cfg_t>
+void seqan2_align_pairwise_edit_distance_benchmark(benchmark::State & state,
+                                                   sequence_pair_generator_t sequence_pair_generator,
+                                                   align_cfg_t seqan3_align_cfg)
+{
+    using seqan3::test::edit_distance_algorithm_seqan2;
+
+    auto [sequences1, sequences2] = sequence_pair_generator(state);
+    constexpr bool is_collection = sequence_pair_generator.is_collection;
+    using sequence_t = typename sequence_pair_generator_t::sequence_t;
+
+    auto algorithm_seqan2 = edit_distance_algorithm_seqan2::select<sequence_t, sequence_t>(seqan3_align_cfg);
+
+    int64_t total = 0;
+    for (auto _ : state)
+    {
+        if constexpr (is_collection)
```

Comment:
Again, please swap the logic here...
```cpp
if constexpr (is_collection)
{
    for (auto _ : state) ...
}
else
{
    for (auto _ : state) ...
}
```

Reply:
Okay I can change it to have the semantics that this are different benchmarks